base/set.jl

```diff
@@ -201,6 +201,109 @@ function unique(f::Callable, C)
     out
 end
 
+# If A is not sorted, then we will need to keep track of all of the elements that we have
+# seen so far.
+function _unique!(A)
+    seen = Set{eltype(A)}()
+    idxs = eachindex(A)
+    i = state = start(idxs)
+    for x in A
+        if x ∉ seen
+            push!(seen, x)
+            i, state = next(idxs, state)
+            A[i] = x
+        end
+    end
+    resize!(A, i - first(idxs) + 1)
+end
+
+# If A is sorted, then we only need to keep track of one element at a time. We replace the
+# elements of A with the unique elements that we see in the order that we see them. Once
+# we have iterated through A, we resize A based on the number of unique elements that we
+# see.
+function _sortedunique!(A)
+    idxs = eachindex(A)
+    y = first(A)
+    state = start(idxs)
+    j, state = next(idxs, state)
+    for i in idxs
+        x = A[i]
+        if !isequal(x, y)
+            j, state = next(idxs, state)
+            y = A[j] = x
+        end
+    end
+    resize!(A, j - first(idxs) + 1)
+end
+
+"""
+    unique!(A::AbstractVector; sort=false)
+
+Remove duplicate items as determined by [`isequal`](@ref), then return the modified `A`.
+By default, `unique!` will return the elements of `A` in the order that they occur. If
+you want `unique!` to return the data sorted, then set `sort=true`. With `sort=true`,
+`unique!` will be able to run faster, however you must ensure that the data can be sorted.
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia> A = [7, 3, 2, 3, 7, 5];
+
+julia> unique!(A)
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+
+julia> A
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+
+julia> B = [7, 6, 42, 6, 7, 42];
+
+julia> unique!(B, sort=true)
+3-element Array{Int64,1}:
+ 6
+ 7
+ 42
+
+julia> B
+3-element Array{Int64,1}:
+ 6
+ 7
+ 42
+```
+"""
+function unique!(A::Union{AbstractVector{<:Real}, AbstractVector{<:AbstractString},
+                          AbstractVector{<:Symbol}}; sort=false)
```

Comment:
Oh, by going through the "outdated" comments, I see that @nalimilan objected that it is like this "since `issorted([1, "a"])` throws an error", but I don't understand, as the type of `[1, "a"]` in not a `AbstractVector{<:Union{Real,AbstractString,Symbol}}`